### PR TITLE
fix selectors

### PR DESCRIPTION
**Pull request by HAOYUatHZ**

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit.rs [hunk 1]
```diff
@@ -434,6 +438,8 @@ pub mod test {
             })
             .collect();
         let (active_gate_rows, active_lookup_rows) = EvmCircuit::get_active_rows(&block);
+        let mut block = block;
+        block.step_num_with_pad = ((1 << k) - 64) / STEP_HEIGHT;
```
Comment (DreamWuGit) [inline zkevm-circuits/src/evm_circuit.rs:442]: want to know why subtract  64 here.

**Comment by DreamWuGit** [comment]
I have a question:  since the `k` value is taken from maximum. of those(fixed table length,  bytecodes length, tx steps' height), the first one we can consider it fixed, the last twos are dynamically per block,  how to make the selectors are same for all blocks ?

File: zkevm-circuits/src/evm_circuit.rs (hunk 1 above)
Comment (lispc) [reply zkevm-circuits/src/evm_circuit.rs:442]: yes.   blinding rows is almost MAX(query of of each gate) + 3.   64 is bigger than it now. 

We 'd better avoid this hardcode value later.

**Comment by lispc** [comment]
> I have a question: since the `k` value is taken from maximum. of those(fixed table length, bytecodes length, tx steps' height), the first one we can consider it fixed, the last twos are dynamically per block, how to make the selectors are same for all blocks ? My understanding of this PR aims to fix selectors for each block, Correct me if I wrong :)

vk is same for same k. Different for different k

**Comment by DreamWuGit** [comment]
For `q_step` , different tx may have different steps,  I assume this approach works in the condition that block only contains one same kind tx with same steps, right ?

**Comment by lispc** [comment]
> For `q_step` , different tx may have different steps, I assume this approach works in the condition that block only contains one same kind tx with same steps, right ?

q_step becomes true every 16 rows. Not related to which opcode/step is.

**Comment by DreamWuGit** [comment]
> 



> q_step becomes true every 16 rows. Not related to which opcode/step is.

for example, tx1 results in 100 steps, tx2 results in 200 steps, hence the first tx`q_step` column contains 100 trues, while the second  tx`q_step` column contains 200 trues, I mean the number of selector values not the same,  right ?

**Comment by lispc** [comment]
ok i understand the question now.  not 100 true or 200 true,  https://github.com/scroll-tech/zkevm-circuits/pull/142/files#diff-191c4ac110f85f0e7c99e4387fabeb9041751b7aacae9f8b11dc7bcc1ae9893cR574 

they are enabled in some fixed cells, every 16 rows til the. bottom of the cells

**Pull request closed without merging**